be/src/fs/hdfs/fs_hdfs.cpp

```diff
@@ -461,7 +461,7 @@ StatusOr<std::unique_ptr<WritableFile>> HdfsFileSystem::new_writable_file(const
 
     hdfsFile file = hdfsOpenFile(hdfs_client->hdfs_fs, path.c_str(), flags, hdfs_write_buffer_size, 0, 0);
     if (file == nullptr) {
-        return Status::InternalError(fmt::format("hdfsOpenFile failed, file={}", path));
```

Comment:
What if user configure wrong credentials, will it be retried also?
Can you just handle 404 situation?

Reply:
https://github.com/apache/hadoop/blob/9acf462d268a7dee8d031978add0034fcf77f861/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs/exception.c#L41

Reply:
https://github.com/apache/hadoop/blob/9acf462d268a7dee8d031978add0034fcf77f861/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs/hdfs.c#L4263

Reply:
just some references.

Reply:
good catch